### PR TITLE
FEATURE: use GitHub API for granting badges instead of cloning repos

### PR DESCRIPTION
**Pull request by OsamaSayegh**

### CODE REVIEW

**Comment by robbiet480** [comment]
@OsamaSayegh One request: Now that it's API based and doesn't require clones, it would be great to support the same syntax as the linkback feature to define what repos to watch. I mostly just want that to be able to monitor `home-assistant/*` instead of having to list all 100 or however many repos we have over at @home-assistant.

**Comment by SamSaffron** [comment]
@majakomel  can you review this PR ?

File: app/lib/github_badges.rb [hunk 1]
```diff
@@ -1,149 +1,119 @@
...
-      @emails = emails
-      @badges = []
-    end
+module DiscourseGithubPlugin
+  module GithubBadges
+    BADGE_NAME_BRONZE ||= 'Contributor'.freeze
+    BADGE_NAME_SILVER ||= 'Great contributor'.freeze
+    BADGE_NAME_GOLD   ||= 'Amazing contributor'.freeze
+
+    COMMITER_BADGE_NAME_BRONZE ||= 'Committer'.freeze
```
Comment (SamSaffron) [inline app/lib/github_badges.rb]: note ... no need for `.freeze` here `# frozen_string_literal: true`

File: app/lib/commits_populator.rb [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      end
+    rescue Octokit::Error => err
+      Rails.logger.warn("#{err.class}: #{err.message}")
+    end
+
+    private
+
+    def is_pr?(commit)
+      commit.commit.author.name != commit.commit.committer.name && commit.commit.committer.name != "GitHub"
+      # GitHub has a special account that acts as the committer of commits that are
```
Comment (SamSaffron) [inline app/lib/commits_populator.rb]: good use of comment here

**Comment by SamSaffron** [comment]
I say do the minor fixed to remove the `.freeze`s and go ahead and merge.

**Pull request merged**